text/3485-feature-documentation.md

```diff
@@ -0,0 +1,195 @@
+- Feature Name: feature-documentation
+- Start Date: 2023-09-09
+- RFC PR: [rust-lang/rfcs#3485](https://github.com/rust-lang/rfcs/pull/3485)
+- Rust Issue:
+  [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC describes a new key to under `features` in `Cargo.toml` for
+documentation. This will allow Cargo to display this information to the user and
+provide a way for `rustdoc` to eventually render this data (how this is rendered
```

Comment:
One of the issues is how rustdoc consumes the data. rust doc generally knows nothing about `Cargo.toml`. I would suggest taking https://github.com/rust-lang/rfcs/pull/3123 as a reference to start a discussion on cargo-rustdoc integration of this. It doesn't need to be perfect but at least two teams should have some consensus.

Reply:
I did bring it up when I initially proposed this feature, https://rust-lang.zulipchat.com/#narrow/stream/266220-rustdoc/topic/Descriptions.20for.20feature.20flags and then opened a draft RFC suggesting that `rustdoc`  accept JSON configuration, which Cargo could pass it https://github.com/rust-lang/rfcs/pull/3421. That didn't get too much traction, though. I will start that discussion back up